### DISCUSSION TITLE
Using a unicorn view as a view

### DISCUSSION THREAD

**Discussion by Joetib** [django-commons/django-unicorn#255] [Ideas]
I find myself writing duplicate views. A Django regular view that simply serves an html which does nothing but load a unicorn view and then the unicorn view itself.

I was wondering if it was possible to present the unicorn view directly to the user.

Also I was wondering if unicorn will have an equivalent of some client side routing strategy

**Comment by adamghill** [comment]
I just attached a draft PR with how I think rendering a component view directly might work.

Something like:

```urls.py
from example.unicorn.components.text_inputs import TextInputsView

urlpatterns = [
    path("direct-view", TextInputsView.as_view(), name="direct-view",),
]
```

How does that look to you?

I still need to figure out how the csrf token and javascript portion would get included on the page. I think that the component would extend/include a template with `{% csrf_token %}` and `{% unicorn_scripts %}`? Does that seem like a reasonable requirement?

**Reply by nerdoc** [reply]
Just for anyone who wonders, the PR is [here](https://github.com/adamghill/django-unicorn/pull/181).

**Comment by Joetib** [comment]
Yeah extending seems great to me.

**Comment by Schroedingberg** [comment]
Is there anything I can do to help to get this one merged soon? My team is using unicorn for a larger project and this would really help us save some development time and keep a cleaner codebase.

**Reply by adamghill** [reply]
Hey @Schroedingberg, let me take a look at this PR again. I don't think I ever figured a super clean way to get this working, but I'll take a look again and let you know.

**Reply by adamghill** [reply]
This feature has been added to 0.40.0 and documentation is at https://www.django-unicorn.com/docs/direct-view/.

**Comment by adamghill** [comment]
https://github.com/adamghill/django-unicorn/pull/181 now works as expected with a basic test case I've set up. I'm going to do some more testing, but am hopeful to merge this into master and tag a new release in the next day or so. The basic gist is that the component template can look like any other Django template (i.e. it can extend a base template, have blocks, etc), but there are two requirements:
- there needs to be an element that wraps the component functionality
- the wrapping element must specify a `unicorn:view` attribute to "tag" that it should be considered the root component
